src/openeo_processes/cubes.py

```diff
@@ -1929,4 +1937,267 @@ def exec_xar(data, process, dimension, target_dimension=None, context=None):
+
+###############################################################################
+# Aggregate_spatial process
+###############################################################################
+
+@process
+def aggregate_spatial():
+
+    return AggregateSpatial()
+
+
+class AggregateSpatial:
+
+    @staticmethod
+    def exec_xar(data, geometries, reducer, target_dimension = "result", context = None):
+
+        if 'x' in data.dims:
+            x = data['x'].values
+        if 'y' in data.dims:
+            y = data['y'].values
+        #if len(data.dims) > 3:
+        #    raise Exception('TooManyDimensions')
+        if type(geometries) == dict:
+            if geometries['type'] == 'MultiPolygon':
+                x_list = []
+                y_list = []
+                for i, p in enumerate(geometries['coordinates']):
+                    if np.shape(p) == (2,):
+                        idx = (np.abs(x - p[0])).argmin()
+                        idy = (np.abs(y - p[1])).argmin()
+                        d_xy = data.sel(x=[x[idx]]).sel(y=[y[idy]])
+                        if callable(reducer):
+                            d_xy = reducer(d_xy, dimension=['x', 'y'])
+                        d_xy[target_dimension] = i
+                        d_xy = d_xy.expand_dims(target_dimension)
+                        if i == 0:
+                            d0 = d_xy
+                        else:
+                            d0 = xr.concat([d0, d_xy], dim = target_dimension)
+                    #elif np.shape(p) == (2,2):
+                        # Line
+                    #elif (np.shape(p))[0] > 2:
+                        # Polygon
+            return d0
+
+
+###############################################################################
+# FitRegrRandomForest process
+###############################################################################
+
+@process
+def fit_regr_random_forest():
+
+    return FitRegrRandomForest()
+
+
+class FitRegrRandomForest:
+
+    @staticmethod
+    def exec_xar(predictors, target, training, num_trees = 100, mtry = None, client = None):
+        training = training/100
+        params = {
+            'learning_rate': 1,
+            'max_depth': 5,
+            'num_parallel_tree': num_trees,
+            'objective': 'reg:squarederror',
+            'subsample': 0.8,
+            'tree_method': 'hist'}
+        if mtry is not None:
+            params['colsample_bynode'] = mtry
+        C = list(target.dims)
+        if 'bands' in C:
+            C.remove('bands')
+        P = list(predictors.dims)
+        if 'bands' in P:
+            P.remove('bands')
+        if P == C:
+            check = np.array([])
+            for dim in P:
+                check = np.append(check, (target[dim].values == predictors[dim].values).all())
+            if (check == False).any():
+                raise Exception('DimensionMismatch')
+        stacked = target.stack(z=C)
+        stacked2 = predictors.stack(z=C)
```

Comment:
I think this wants to be `stacked2 = predictors.stack(z=C)
```suggestion
        stacked2 = predictors.stack(z=P)
```